src/main/java/org/kitteh/irc/client/library/Client.java

```diff
@@ -321,6 +331,83 @@
     }
 
     /**
+     * Proides commands.
+     */
+    interface Commands {
+        /**
+         * Provides a new AWAY command.
+         *
+         * @return new away command
+         */
+        @Nonnull
+        AwayCommand away();
+
+        /**
+         * Provides a new CAP REQ command.
+         *
+         * @return new capability request command
+         */
+        @Nonnull
+        CapabilityRequestCommand capabilityRequest();
+
+        /**
+         * Provides a new MODE command.
+         *
+         * @return new mode command
+         */
+        @Nonnull
+        ChannelModeCommand mode(@Nonnull Channel channel);
+
+        /**
+         * Provides a new KICK command.
+         *
+         * @return new kick command
+         */
+        @Nonnull
+        KickCommand kick(@Nonnull Channel channel);
+
+        /**
+         * Provides a new MONITOR command.
+         *
+         * @return new monitor command
+         */
+        @Nonnull
+        MonitorCommand monitor();
+
+        /**
+         * Provides a new OPER command.
+         *
+         * @return new oper command
+         */
+        @Nonnull
+        OperCommand oper();
+
+        /**
+         * Provides a new TOPIC command.
+         *
+         * @return new topic command
+         */
+        @Nonnull
+        TopicCommand topic(@Nonnull Channel channel);
+
+        /**
+         * Provides a new WALLOPS command.
+         *
+         * @return new wallops command
+         */
+        @Nonnull
+        WallopsCommand wallops();
```

Comment:
I think people will mis-use certain commands if they see it. When it's hidden from view only people actively looking for it will find it. As a developer I'd prefer things not being shown to me if it is only relevant to a subset of a subset of users - it's clutter I won't use and am more likely to misuse by thinking it is for something else. (For instance, some people think WALLOPS are STATUSMSGes, like how hexchat(?) has a /wallopchan command which is a all-connected-network-wide AMSG.

Consistency being the problem, I'd prefer it being abstracted to a separate client with the specific intention of using elevated-level commands. (Also, just a bit of paranoia. I just don't trust many users to deal with things they don't understand)